reference/conanfile/attributes.rst

```diff
@@ -965,7 +998,23 @@ You can also model system dependencies for each component or just header files.
     The information of components is not lost but aggregated to the *global* scope and the usage of components should be transparent right
     now to consumers and generators.
 
-The structure of the ``Component`` object is the same as the one used by the ``cpp_info`` object and has **the same default directories**.
+This syntax is also prepared (still not implemented in generators) to make a component require a complete external package or a component
+from a different package using the special character ``::``. For example, here we have a recipe that requires ``zlib`` and ``OpenSSL``, and
+it has two components: ``comp1`` requires the whole ``zlib`` package and ``comp2`` requires ``comp1`` as well as the ``ssl`` component in
+the ``OpenSSL`` package.
+
+.. code-block:: python
+
+    class MyConan(ConanFile):
+        ...
+        requires = "zlib/1.2.11", "openssl/1.1.1g"
+
+    def package_info(self):
+        self.cpp_info.components["comp1"].requires("zlib::zlib")
+        self.cpp_info.components["comp2"].requires("comp1", "openssl::ssl")
+
+Note that component requires should **explicitly** define requirements to other package (``zlib::zlib``) or other package components
+(``openssl::ssl``).
```

Comment:
I would convert this into a "note", "important", or similar, so it is highlighted in the documentation. Even though Conan will raise an error, for me it is very important, it the only difference between the top-level `cpp_info` and the components' one